### PR TITLE
docs(js): more passwordless sign in examples

### PR DESCRIPTION
**Pull request by jjarvisp**
#### Description of changes:
The purpose of this pull request is to expand existing passwordless sign in examples to better illustrate the available options.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [X] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [X] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [X] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx [hunk 1]
```diff
@@ -1120,17 +1120,28 @@ Your application's users can also sign in using passwordless methods. To learn m
...
-To request an OTP code via SMS for authentication, the challenge is passed as the challenge response to the confirm sign in API.
+Pass `SMS_OTP` as the `preferredChallenge` when calling the `signIn` API in order to initiate a passwordless authentication flow with SMS OTP.
 
-Amplify will respond appropriately to Cognito and return the challenge as sign in next step: `CONFIRM_SIGN_IN_WITH_SMS_CODE`:
 
 ```ts
-const { nextStep } = await confirmSignIn({
-    challengeResponse: "SMS_OTP"
+const { nextStep: signInNextStep } = await signIn({
+	username: 'example-username',
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx:1128]:
```suggestion
	username: '+15551234567',
```

File: src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx [hunk 2]
```diff
@@ -1223,17 +1234,27 @@ func confirmSignIn() -> AnyCancellable {
...
-To request an OTP code via email for authentication, the challenge is passed as the challenge response to the confirm sign in API.
-
-Amplify will respond appropriately to Cognito and return the challenge as sign in next step: `CONFIRM_SIGN_IN_WITH_EMAIL_CODE`:
+Pass `EMAIL_OTP` as the `preferredChallenge` when calling the `signIn` API in order to initiate a passwordless authentication flow using email OTP.
 
 ```ts
-const { nextStep } = await confirmSignIn({
-    challengeResponse: "EMAIL_OTP"
+const { nextStep: signInNextStep } = await signIn({
+	username: 'example-username',
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx:1241]:
```suggestion
	username: 'hello@example.com',
```

File: src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx [hunk 3]
```diff
@@ -1326,15 +1347,20 @@ func confirmSignIn() -> AnyCancellable {
...
 <InlineFilter filters={["angular", "javascript", "nextjs", "react", "react-native", "vue"]}>
 
-The WebAuthn credential flow is initiated by passing the challenge name to the confirm sign in api.
+Pass `WEB_AUTHN` as the `preferredChallenge` in order to initiate the passwordless authentication flow using a WebAuthn credential.
 
 ```ts
-const { nextStep } = await confirmSignIn({
-    challengeResponse: "WEB_AUTHN",
+const { nextStep: signInNextStep } = await signIn({
+	username: 'example-username',
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx:1354]:
```suggestion
	username: 'hello@example.com',
```

File: src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx [hunk 4]
```diff
@@ -1356,25 +1382,52 @@ handleNextSignInStep(nextStep);
...
+### Password
 
-Traditional password based authentication is available from this flow as well. To initiate this flow from select challenge, either `PASSWORD` or `PASSWORD_SRP` is passed as the challenge response.
+ Pass either `PASSWORD` or `PASSWORD_SRP` as the `preferredChallenge` in order to initiate a traditional password based authentication flow.
 
 ```ts
-const { nextStep } = await confirmSignIn({
-    challengeResponse: "PASSWORD_SRP", // or "PASSWORD"
+const { nextStep: signInNextStep } = await signIn({
+	username: 'example-username',
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx:1391]:
```suggestion
	username: 'hello@example.com',
```

File: src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx [hunk 5]
```diff
@@ -1356,25 +1382,52 @@ handleNextSignInStep(nextStep);
...
-    challengeResponse: "Test123#",
+### First Factor Selection
+
+Omit the `preferredChallenge` parameter to discover what first factors are available for a given user. 
+
+The `confirmSignIn` API can then be used to select a challenge and initiate the associated authentication flow.
+
+```ts
+const { nextStep: signInNextStep } = await signIn({
+	username: 'example-username',
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/connect-your-frontend/sign-in/index.mdx:1413]:
```suggestion
	username: '+15551234567',
```

**Comment by josefaidt** [comment]
Hey @jjarvisp left a few comments to adjust the username entries to emails and phone numbers, but overall LGTM!

**Review by josefaidt — APPROVED** [review]

**Review by yuhengshs — APPROVED** [review]

**Pull request merged**